flowjax/bijections/bijection.py

```diff
@@ -148,6 +208,33 @@ def inverse_and_log_det(
                 Required for conditional bijections. Defaults to None.
         """
 
+    def inverse_gradient_and_val(
+        self,
+        y: Array,
+        y_grad: Array,
+        y_logp: Array,
+        condition: Array | None = None,
+    ) -> tuple[Array, Array, Array]:
+        """Map position and logp gradient through inverse transformation.
+
+        Args:
+            y: Point in the transformed space, with shape matching bijection.shape.
+            y_grad: Gradient of the log density on the transformed space, with the
+                same shape as y.
+            y_logp: log density on the transformed space.
+            condition: Condition array with shape matching bijection.cond_shape.
+                Required for conditional bijections. Defaults to None.
+
+        Returns:
+            x: Point on the untransformed space, ie `bjiection.inverse(x)`.
+            x_grad: The gradient of the log density at `x`.
+            x_logp: The log density on the untransformed space at `x`.
+        """
+        x, logdet = self.inverse_and_log_det(y, condition)
```

Comment:
I assume you don't need the log_det in the inverse?
```python
x = bijection.inverse(y, condition)
(_, fwd_log_det), pull_grad_fn = jax.vjp(...)
```
and add ``fwd_log_det`` instead of subtracting the inverse log det.

Reply:
Thanks, that is better.